src/Cake/Program.cs

```diff
@@ -96,6 +96,7 @@ private ITypeRegistrar BuildTypeRegistrar()
             builder.RegisterType<InfoFeature>().As<ICakeInfoFeature>().SingleInstance();
 
             // Core
+            builder.RegisterType<LaunchInfo>().As<ILaunchInfo>().SingleInstance();
```

Comment:
This registration looks good, but it doesn't actually causes `LaunchInfo` to be instantiated at this time - it will only be instantiated later when some dependency needs it (i.e. `CakeEnvironment`).

Now, this _might_ work currently because the `CakeEnvironment` is instantiated early in the pipeline which should cause the `LaunchInfo` to be instantiated early enough as well - but it's not 100% guaranteed it will continue like that in the future (e.g. if we introduce other dependencies)

I'd prefer to play safe here and create the instance of `LaunchInfo` ourselves at the very beginning, and register it into the container, so that know for sure that we captured the `CurrentDirectory` at the right time.

e.g.
```csharp
private ITypeRegistrar BuildTypeRegistrar()
{
    var builder = new ContainerBuilder();

    var launchInfo = new LaunchInfo(); // We're 100% sure we captured the working directory within `launchInfo`
    builder.RegisterInstance(launchInfo).As<ILaunchInfo>();

     // ...
```